netty/src/main/java/io/grpc/netty/UdsNameResolverProvider.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.netty;
+
+import com.google.common.base.Preconditions;
+import io.grpc.NameResolver;
+import io.grpc.NameResolverProvider;
+import java.net.URI;
+
+public class UdsNameResolverProvider extends NameResolverProvider {
```

Comment:
It seems `getProducedSocketAddressTypes()` is missing. There needs to be a higher-level test that actually combines the name resolver and channel provider together, because right now this name resolver would only ever fail.